Edit the channel visibility of products on the Product Feed page

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

This PR based on the relevant backend API implementation #655

Closes #627 

- Add functions to `.~/data` for updating the channel visibility of products
- Implement the action component for edit product visibility and make this feature able to use on the Product Feed page

### Screenshots:

![Kapture 2021-05-25 at 17 26 50](https://user-images.githubusercontent.com/17420811/119474344-a1063680-bd7e-11eb-9d5b-9fab3e1dbb2a.gif)

### Detailed test instructions:

1. Go to the product feed page: `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed`
2. Hover on the edit icon, and then a tooltip should show up with "Select one or more products to bulk edit"
3. Select 1 or more products, the edit icon should change to a dropdown selection and a disabled action button
4. Hover on the action button, and a tooltip should show up with "Select channel visibility" if you haven’t selected channel visibility yet
5. Select "Sync and show" or "Don’t sync and show" by the dropdown selection, the action button should be able to click
6. Click on the action button, then the channel visibility of selected products should be updated
7. The data of "Issue to resolve" and "Product Feed" tables should be re-fetched to reflect the updates.
8. A successful message should be displayed if the updates called successfully

### Changelog Note:

> Edit the channel visibility of products on the Product Feed page.

eason9487 committed:
Add functions to  for updating the visibility of products

eason9487 committed:
Implement the action component of edit product visibility

eason9487 committed:
Implement the feature of edit product visibility on the Product Feed page

eason9487 committed:
Display successful message after updating product visibility

j111q reviewed (approved):
Yup, perfect thanks @eason9487!

eason9487 committed:
Fix the react warning of invalid prop type

Address https://github.com/woocommerce/google-listings-and-ads/pull/662#issuecomment-848607983

tomalec reviewed (approved):
Reviewed the code, tested locally, LGTM
